lib/config/rule-validator.js

```diff
@@ -63,7 +68,9 @@ function throwRuleNotFoundError({ pluginName, ruleName }, config) {
         // falls through to throw error
     }
 
-    throw new TypeError(errorMessage);
+    const finalError = errorMessage === "" ? error : new TypeError(errorMessage);
+
```

Comment:
This isn't necessary if we only set `errorMessage` before the `if` statements.

Reply:
If i summarize all three suggestions then it looks something like this i think: 

```js
let errorMessage = `${errorMessageHeader}: Could not find plugin "${pluginName}" in configuration.`;

if (config.plugins && config.plugins[pluginName]) {
errorMessage = 'some other error message';
}

const error = new TypeError(errorMessage);

error.messageTemplate = "config-plugin-missing";
error.messageData = { pluginName, ruleId };

throw error;
```

but won't it effect the error message assigned within the `if` statement?

Reply:
You're talking about `messageTemplate`? Yes, you're right, we probably want to clear its value if the message changes.

Reply:
Done!